packages/design-system/modal-screen/with-modal-screen.tsx

```diff
@@ -20,15 +24,34 @@ function withModalScreen<P extends {}>(
   // eslint-disable-next-line react/display-name
   return function (props: P) {
     const [title, setTitle] = useState(titleProp);
+    const closeCallback = useRef<PopCallback>(null);
+
+    const [contextValues, setContextValues] = useState<ModalScreenContextValue>(
+      {
+        setTitle,
+        pop: () => {},
+      }
+    );
     const modalRef = useRef<ModalMethods>(null);
     useBackPressHandler(modalRef, backPressHandlerEnabled);
 
     const router = useRouter();
     const onClose = useCallback(() => {
       router.pop();
+      closeCallback.current?.();
     }, [router]);
 
-    const contextValues = useMemo(() => ({ setTitle }), []);
+    useLayoutEffect(() => {
+      if (modalRef.current?.close) {
+        setContextValues({
+          setTitle,
+          pop: (params) => {
+            closeCallback.current = params?.callback;
```

Comment:
I added this method, which is copied from the creator-channels branch code, It's useful to do some callback when you want to wait for the modal show/hide animation to be dismissed.